arch/arm64/core/CMakeLists.txt

```diff
@@ -15,6 +15,18 @@ zephyr_library_sources(
   vector_table.S
 )
 
+# Use large code model for addresses larger than 32 bits,
+# This is 10 characters long (0x12345678). We can't use a
+# simple numeric comparison because these values may be
+# beyond the numeric range of integers for cmake.
+
+string(LENGTH "x${CONFIG_SRAM_BASE_ADDRESS}" SRAM_LENGTH)
+string(LENGTH "x${CONFIG_KERNEL_VM_BASE}" KERNEL_VM_LENGTH)
+
+if(${SRAM_LENGTH} GREATER 11 OR ${KERNEL_VM_LENGTH} GREATER 11)
+  zephyr_cc_option(-mcmodel=large)
+endif()
```

Comment:
Is there any issue to have this always set for ARM64?

Reply:
I found this:
```
       -mcmodel=tiny
           Generate code for the tiny code model.  The program and its statically defined symbols must be within 1MB of each other.  Programs can be statically or dynamically linked.

       -mcmodel=small
           Generate code for the small code model.  The program and its statically defined symbols must be within 4GB of each other.  Programs can be statically or dynamically linked.  This is the default code model.

       -mcmodel=large
           Generate code for the large code model.  This makes no assumptions about addresses and sizes of sections.  Programs can be statically linked only.
```

Reply:
AArch64 Options in GCC

Reply:
Right, but Zephyr is by default only statically linked so why not set it all the times?

Reply:
Yes, I think no problem

Reply:
Presumably there's a code size impact; the relocations become physically larger and presumably require different instruction formats.

Reply:
Yes, the generated code is larger and slower. But, it's also going to work, while the other models may have distant relocations which the linker will fail to resolve. What we want is a way to tell the compiler that *code* references are near and *data* references may be far. Not sure that's one of the options.

Reply:
alright I propose to go with this for now.